src/iface/ethernet.rs

```diff
@@ -41,24 +42,28 @@ impl<'a, 'b, 'c, DeviceT: Device + 'a> Interface<'a, 'b, 'c, DeviceT> {
     /// # Panics
     /// See the restrictions on [set_hardware_addr](#method.set_hardware_addr)
     /// and [set_protocol_addrs](#method.set_protocol_addrs) functions.
-    pub fn new<DeviceMT, ArpCacheMT, ProtocolAddrsMT>
+    pub fn new<DeviceMT, ArpCacheMT, ProtocolAddrsMT, DefaultGWMT>
```

Comment:
Nit: I think `MT` used to mean something but now it can be just replaced with `T` everywhere here (so `GatewayT`).